robot/board.py

```diff
@@ -15,10 +15,9 @@ class Board:
 
     def __init__(self, socket_path: Union[Path, str]) -> None:
         self.socket_path = Path(socket_path)
-        self.socket = None
-        self.data = b''
 
-        self._connect()
+        self.connection = Connection(self._get_socket())
```

Comment:
This code is also duplicated in `_reconnect`. Perhaps it would be neater to initialise `connection` to `None` here and then immediately call `_reconnect`, and modify `_reconnect` to add a check for `connection is not None` around the call to `connection.close`?

Reply:
This is deliberate -- I'm precisely moving away from having `None`s floating around the place. This is partly because they're annoying and partly because adding types exposes that.

Reply:
Makes sense. In that case, it could be factored out into a new method that returns the new `Connection`?

Reply:
e.g.

```python
def _connect(self):
  new_connection = Connection(self._get_socket())
  self._greeting_response(new_connection.receive())
  return new_connection

def __init__(self):
  # ...
  self.connection = self._connect()

def _reconnect(self):
  self.connection.close()
  self.connection = self._connect()
```

Assuming `self.connection` is not modified outside of the above functions, it will never be `None`.

In theory `_greeting_response` would need to be modified Assuming the above code is the olAssuming the above code is the olto use `new_connection` rather than `self.connection`, though since it's now only used once it could just be inlined into `_connect`. However I've just seen that our implementation of `_greeting_response` is simply a no-op, so I think it'd be clearer to just delete it (since clearly robotd does not require nor accept a response to the greeting packet it sends).

Reply:
`_greeting_response` is, I _think_, provided so that `Board` implementations can handle the initial status broadcast from `robotd`. You're right that it's unused at the moment.

I'll have a look at the rest later.